banking/public/js/bank_reconciliation_beta/actions_panel/match_tab.js

```diff
@@ -16,19 +16,31 @@ erpnext.accounts.bank_reconciliation.MatchTab = class MatchTab {
 		});
 		this.match_field_group.make()
 
-		this.summary_empty_state();
 		await this.populate_matching_vouchers();
 	}
 
 	summary_empty_state() {
-		let summary_field = this.match_field_group.get_field("transaction_amount_summary").$wrapper;
-		summary_field.append(
-			`<div class="report-summary reconciliation-summary" style="height: 90px;">
-			</div>`
+		this.render_transaction_amount_summary(0, 0, 0, this.transaction.currency);
+	}
+
+	matching_vouchers_empty_state() {
+		this.match_field_group.get_field("vouchers").$wrapper.empty();
+		this.match_field_group.get_field("vouchers").$wrapper.append(
+			`<div class="text-muted text-center mb-4">${__("Loading ...")}</div>`
 		);
 	}
 
-	async populate_matching_vouchers() {
+	async populate_matching_vouchers(event_obj) {
+		if (event_obj && event_obj.type === "input") {
+			// `bind_change_event` in `data.js` triggers both an input and change event
+			// This triggers the `populate_matching_vouchers` twice on clicking on filters
+			// Since the input event is debounced, we can ignore it for a checkbox
+			return;
+		}
+
+		this.summary_empty_state();
+		this.matching_vouchers_empty_state();
```

Comment:
```suggestion
		this.render_data_table([]);
		this.actions_table.freeze();
```

If we do it like this, and also add `freezeMessage: __("Loading ...")` to the `datatable_options`, we can already show the table skeleton during loading:

![Bildschirmfoto 2024-10-16 um 10 51 24](https://github.com/user-attachments/assets/9a8e3aff-4c2d-46a9-81d5-b9ae210e2c0a)

Reply:
I'm not a fan of the aesthetic appeal and that "No Data" is visible. But I'm also not very strongly opinionated here 😄  Thanks for your commit!